Build failed with nodejs >= 12

Hello

```
$ npm i function-origin

> function-origin@1.1.1 install /home/schroeterm/devel/westeros-dev/node_modules/test/node_modules/function-origin
> node-gyp rebuild

make : on entre dans le répertoire « /home/schroeterm/devel/westeros-dev/node_modules/test/node_modules/function-origin/build »
  CXX(target) Release/obj.target/function_origin/src/function_origin.o
In file included from ../src/function_origin.cc:2:
../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../nan/nan.h:2298:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/function_origin.cc: In function ‘Nan::NAN_METHOD_RETURN_TYPE SetOrigin(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/function_origin.cc:11:79: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
   target->Set(Nan::New<String>("file").ToLocalChecked(), origin.ResourceName());
                                                                               ^
In file included from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8-internal.h:14,
                 from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h:25,
                 from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/node.h:63,
                 from ../src/function_origin.cc:1:
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h:3372:22: note: declared here
                 bool Set(Local<Value> key, Local<Value> value));
                      ^~~
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/function_origin.cc:14:49: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     Nan::New<Integer>(fn->GetScriptLineNumber()));
                                                 ^
In file included from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8-internal.h:14,
                 from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h:25,
                 from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/node.h:63,
                 from ../src/function_origin.cc:1:
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h:3372:22: note: declared here
                 bool Set(Local<Value> key, Local<Value> value));
                      ^~~
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/function_origin.cc:17:51: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     Nan::New<Integer>(fn->GetScriptColumnNumber()));
                                                   ^
In file included from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8-internal.h:14,
                 from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h:25,
                 from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/node.h:63,
                 from ../src/function_origin.cc:1:
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h:3372:22: note: declared here
                 bool Set(Local<Value> key, Local<Value> value));
                      ^~~
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/function_origin.cc:19:87: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
   target->Set(Nan::New<String>("inferredName").ToLocalChecked(), fn->GetInferredName());
                                                                                       ^
In file included from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8-internal.h:14,
                 from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h:25,
                 from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/node.h:63,
                 from ../src/function_origin.cc:1:
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h:3372:22: note: declared here
                 bool Set(Local<Value> key, Local<Value> value));
                      ^~~
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/function_origin.cc: At global scope:
../src/function_origin.cc:22:18: error: variable or field ‘Init’ declared void
 static void Init(Handle<Object> exports) {
                  ^~~~~~
../src/function_origin.cc:22:18: error: ‘Handle’ was not declared in this scope
../src/function_origin.cc:22:18: note: suggested alternative: ‘rand_r’
 static void Init(Handle<Object> exports) {
                  ^~~~~~
                  rand_r
../src/function_origin.cc:22:31: error: expected primary-expression before ‘>’ token
 static void Init(Handle<Object> exports) {
                               ^
../src/function_origin.cc:22:33: error: ‘exports’ was not declared in this scope
 static void Init(Handle<Object> exports) {
                                 ^~~~~~~
In file included from ../src/function_origin.cc:1:
../src/function_origin.cc:27:30: error: ‘Init’ was not declared in this scope
 NODE_MODULE(function_origin, Init)
                              ^~~~
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/node.h:560:36: note: in definition of macro ‘NODE_MODULE_X’
       (node::addon_register_func) (regfunc),                          \
                                    ^~~~~~~
../src/function_origin.cc:27:1: note: in expansion of macro ‘NODE_MODULE’
 NODE_MODULE(function_origin, Init)
 ^~~~~~~~~~~
../src/function_origin.cc:27:30: note: suggested alternative:
 NODE_MODULE(function_origin, Init)
                              ^~~~
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/node.h:560:36: note: in definition of macro ‘NODE_MODULE_X’
       (node::addon_register_func) (regfunc),                          \
                                    ^~~~~~~
../src/function_origin.cc:27:1: note: in expansion of macro ‘NODE_MODULE’
 NODE_MODULE(function_origin, Init)
 ^~~~~~~~~~~
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/node.h:217:18: note:   ‘node::Init’
 NODE_EXTERN void Init(int* argc,
                  ^~~~
In file included from /home/schroeterm/.cache/node-gyp/12.10.0/include/node/node.h:63,
                 from ../src/function_origin.cc:1:
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]’:
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/node_object_wrap.h:84:78:   required from here
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h:9893:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<node::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]’:
../../nan/nan_object_wrap.h:65:61:   required from here
/home/schroeterm/.cache/node-gyp/12.10.0/include/node/v8.h:9893:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
make: *** [function_origin.target.mk:109: Release/obj.target/function_origin/src/function_origin.o] Error 1
make : on quitte le répertoire « /home/schroeterm/devel/westeros-dev/node_modules/test/node_modules/function-origin/build »
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/schroeterm/.npm-packages/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:196:23)
gyp ERR! stack     at ChildProcess.emit (events.js:209:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Linux 4.19.0-6-amd64
gyp ERR! command "/usr/bin/node" "/home/schroeterm/.npm-packages/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/schroeterm/devel/westeros-dev/node_modules/test/node_modules/function-origin
gyp ERR! node -v v12.10.0
gyp ERR! node-gyp -v v5.0.3
gyp ERR! not ok 
npm WARN test@1.0.0 No description
npm WARN test@1.0.0 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! function-origin@1.1.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the function-origin@1.1.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/schroeterm/.npm/_logs/2019-10-01T12_32_59_341Z-debug.log
```

--- a/src/function_origin.cc
+++ b/src/function_origin.cc
@@ -3,7 +3,8 @@
 
 using namespace v8;
 
-NAN_METHOD(SetOrigin) {
+NAN_METHOD(SetOrigin)
+{
   Local<Function> fn = info[0].As<Function>();
   Local<Object> target = info[1].As<Object>();
   ScriptOrigin origin = fn->GetScriptOrigin();
@@ -11,17 +12,19 @@ NAN_METHOD(SetOrigin) {
   target->Set(Nan::New<String>("file").ToLocalChecked(), origin.ResourceName());
 
   target->Set(Nan::New<String>("line").ToLocalChecked(),
-    Nan::New<Integer>(fn->GetScriptLineNumber()));
+              Nan::New<Integer>(fn->GetScriptLineNumber()));
 
   target->Set(Nan::New<String>("column").ToLocalChecked(),
-    Nan::New<Integer>(fn->GetScriptColumnNumber()));
+              Nan::New<Integer>(fn->GetScriptColumnNumber()));
 
   target->Set(Nan::New<String>("inferredName").ToLocalChecked(), fn->GetInferredName());
 }
 
-static void Init(Handle<Object> exports) {
+static void Init(Local<v8::Object> exports)
+{
+  v8::Local<v8::Context> context = exports->CreationContext();
   exports->Set(Nan::New<String>("SetOrigin").ToLocalChecked(),
-    Nan::New<FunctionTemplate>(SetOrigin)->GetFunction());
+               Nan::New<FunctionTemplate>(SetOrigin)->GetFunction(context).ToLocalChecked());
 }
 
 NODE_MODULE(function_origin, Init)

